### PR TITLE
tests/driver_my9221: fix for loop indexes

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
In the `my9221` test there are some for loops used for testing each led in the driver. However, the indexes should go always from 0 to 9 and the actual test generate these indexes from 1 to 10, so there's a failed assertion [here](https://github.com/RIOT-OS/RIOT/blob/68dc5b0d6e0422df9c3653c2cb8021fc35974aae/drivers/my9221/my9221.c#L133).

This PR fixes this issue.

### Testing procedure
Configure `drivers_my9221` test in its Makefile, and run `make flash test`. Verify there's no `FAILED ASSERTION` => `[SUCCESS]` message in the end.

### CODE REVIEW

File: tests/driver_my9221/main.c [hunk 1]
```diff
@@ -45,16 +45,16 @@ int main(void)
     }
     /* run some tests */
     LOG_INFO("- light up all LEDs one by one.\n");
-    for (unsigned i=0; i < dev.params.leds; ++i) {
+    for (unsigned i=0; i < dev.params.leds; i++) {
```
Comment (cgundogan) [inline tests/driver_my9221/main.c]: this change is unrelated

File: tests/driver_my9221/main.c [hunk 2]
```diff
@@ -45,16 +45,16 @@ int main(void)
...
         my9221_set_led(&dev, i, MY9221_LED_ON);
         xtimer_usleep(TEST_WAIT);
         my9221_set_led(&dev, i, MY9221_LED_OFF);
     }
     xtimer_usleep(TEST_WAIT);
-    for (unsigned i=dev.params.leds; i > 0 ; --i) {
-        my9221_set_led(&dev, i, MY9221_LED_ON);
+    for (unsigned i=dev.params.leds; i > 0 ; i--) {
```
Comment (cgundogan) [inline tests/driver_my9221/main.c]: also unrelated

Comment (cladmi) [reply tests/driver_my9221/main.c]: I would rather change the boundaries, because in the same function `i` was first a led number, then a led +1.
But as it is an unsigned so of `>= 0` cannot be used.

What is our normal way of handling these ?
Iterate on `j` and define `i = j - 1` as an intermediate variable ?

File: tests/driver_my9221/main.c (hunk 1 above)
Comment (jia200x) [reply tests/driver_my9221/main.c]: you are right, late programming effects :)

File: tests/driver_my9221/main.c (hunk 2 above)
Comment (jia200x) [reply tests/driver_my9221/main.c]: that's why I prefer the `while(i--)` pattern for reverse counts :)

But yeah, it's cleaner with an intermediate variable. I will push it.

**Comment by jia200x** [comment]
Done. I squashed since the change is super small.
